tests/test_compile.py

```diff
@@ -102,7 +103,13 @@ def test_compile__compile_restricted_exec__5():
     assert result.code is None
     assert result.warnings == []
     assert result.used_names == {}
-    assert result.errors == ('source code string cannot contain null bytes',)
+    if IS_PY312_OR_GREATER:
+        assert result.errors == (
+            'Line None: SyntaxError: source code string cannot contain null'
+            ' bytes at statement: None',)
+    else:
+        assert result.errors == (
+            'source code string cannot contain null bytes',)
```

Comment:
This should fit all Python versions, if it is okay that we do not test the exact error message:

```suggestion
    assert 'source code string cannot contain null bytes' in result.errors
```

Reply:
`result.errors` is a tuple, so it is not that easy. I'll keep my version.